### PR TITLE
refresh tracking events (bug 1126625)

### PR DESCRIPTION
**Pull request by ngokevin**
PR:

- move some more tracking events into tracking_events.js
- move the tracking logging duties into tracking.js
- fix lightbox events
- pull in modal update that lets it work better on phantom
- remove rickfant rolled egg
- test most tracking events besides app install/launch ones
- enable tracking by default
- test for description toggle and feed navigations

Depends on https://github.com/mozilla/marketplace-api-mock/commit/6ad63d445123deacdc4ba321984db5e451d83b70 which has been committed

### CODE REVIEW

File: src/media/js/views/search.js [hunk 1]
```diff
@@ -99,9 +98,8 @@ define('views/search',
         return query;
     }
 
-    z.body.on('submit', 'form#search', function(e) {
+    z.body.on('submit', 'form#search', _pd(function(e) {
```
Comment (spasovski) [inline src/media/js/views/search.js]: just `#search` is fine

File: src/media/js/views/search.js [hunk 2]
```diff
@@ -149,9 +147,6 @@ define('views/search',
                     data.unshift(base('Wa-pa-pa-pa-pa-pa-pow!'));
```
Comment (spasovski) [inline src/media/js/views/search.js]: thanks for leaving this one in :+1:

File: src/media/js/tracking_events.js [hunk 3]
```diff
@@ -25,128 +30,159 @@ define('tracking_events',
...
+            'App view interactions',
+            'click',
+            'Toggle description'
+        );
     })
 
+    // Lightbox open (previews, screenshots).
+    .on('lightbox-open', function() {
+        if (z.context.type.split(' ').indexOf('detail') !== -1) {
+            tracking.trackEvent(
```
Comment (spasovski) [inline src/media/js/tracking_events.js]: `trackEvent`? Probably a copy-paste but we have it declared above.

Comment (ngokevin) [inline src/media/js/tracking_events.js]: yeh thanks

File: src/media/js/tracking_events.js [hunk 4]
```diff
@@ -164,42 +200,40 @@ define('tracking_events',
...
     }
 
     function track_app_install_begin(product, $install_btn) {
         // Track app install start.
-        track([
+        trackEvent(
             'Click to install app',
             product.receipt_required ? 'paid' : 'free',
-            product.name + ':' + product.id,  // TODO: ask to use product.slug.
+            product.name + ':' + product.id,
```
Comment (spasovski) [inline src/media/js/tracking_events.js]: I agree that this is weird.

File: tests/lib/helpers.js [hunk 5]
```diff
@@ -105,6 +116,41 @@ function checkValidity(selector) {
 }
 
 
+function assertUATracking(test, trackArgs) {
+    // Check that a UA tracking event or variable change was made
+    // by checking the tracking logs.
+    // If trackArgs is a string, it will just check the first argument.
+    var isString = false;
+    if (typeof trackArgs == 'string') {
```
Comment (spasovski) [inline tests/lib/helpers.js]: `trackArgs.constructor === String` would also work

File: tests/lib/helpers.js [hunk 6]
```diff
@@ -105,6 +116,41 @@ function checkValidity(selector) {
 }
 
 
+function assertUATracking(test, trackArgs) {
+    // Check that a UA tracking event or variable change was made
```
Comment (spasovski) [inline tests/lib/helpers.js]: This is an interesting idea indeed.

**Comment by spasovski** [comment]
This is good work as far as being thorough with testing, especially where we test tracking. It does add a bit of complexity to future tracking work but the code is quite readable so onboarding shouldn't be too bad.

r+

**Comment by ngokevin** [comment]
i don't think it adds complexity to tracking code. all events are encapsulated in the tracking_events, and the track_logs are appropriately handled by tracking.js. the tests caught at least 4 tracking_events that weren't firing too. 

thanks

**Comment by spasovski** [comment]
Yea I don't disagree with the usefulness. It was "simpler" before but also more prone to issues where we didn't know if something was tracked. In either case the logic is simple to follow so it will make sense.

**Comment by ngokevin** [comment]
/me thinks it's simpler now, you can see everything we're tracking in one file

**Pull request merged**